test/test_fork.py

```diff
@@ -106,10 +107,27 @@ def action(client):
                     rc = self.clients[i % len(self.clients)]
                     if i % 50 == 0 and self.fork:
                         # Fork
-                        with self.runner.fork() as pid:
-                            if pid == 0:  # Child
-                                for c in self.clients:
-                                    action(c)
+                        def target():
```

Comment:
Let's refactor the  `self.runner.fork()` method to use multiprocessing internally. The body of the with statement would then _only_ be run in the parent process:
```python
def target():
    for c in self.clients:
        action(c)
        c.close()  # Client no longer needed.

with self.runner.fork(target=target) as proc:
    self.assertTrue(proc.pid)
```